ui/vty/src/Ariadne/UI/Vty/Widget.hs

```diff
@@ -383,12 +391,19 @@ withWidgetState action = do
 ----------------------------------------------------------------------------
 
 drawWidget :: WidgetName -> p -> Widget p -> B.Widget WidgetName
-drawWidget focus parent (Widget widget@WidgetInfo{ widgetState, widgetDrawWithFocus }) =
-  runReader (runReaderT (widgetDrawWithFocus focus widgetState) widget) parent
+drawWidget focus parent (Widget widget@WidgetInfo{..}) =
```

Comment:
I don't quite undestand how this works. You are adding visibility requests to all widgets, no matter focused or not?

Reply:
No, I'm adding visibility requests only if widget is focused `(focus == widgetName)` _and_ we don't have `widgetIgnoreVisibility` flag enabled.

Reply:
Crap, I did not notice the bit about the focus. Sorry.